tests/conftest.py

```diff
@@ -176,6 +179,36 @@ def mock_db():
     athena.stop()
 
 
+@pytest.fixture
+def mock_db(tmp_path):
+    def _compat_regexp_like(string: str | None, pattern: str | None) -> bool:
+        match = re.search(pattern, string)
+        return match is not None
+
+    def _compat_from_iso8601_timestamp(
+        value: str | datetime.datetime,
+    ) -> datetime.datetime:
+        if type(value) is str:
+            return datetime.datetime.fromisoformat(value)
+        return value
+
+    db = duckdb.connect(tmp_path / "duck.db")
+    db.create_function(
```

Comment:
Obviously you thought about this, but let's spill a tiny amount of digital ink. Pros/cons of having tests depend on cumulus-library and grabbing its DuckdbDatabase class? Not worth it yet, I'm guessing?

Reply:
Yeah, a couple thoughts

- the aggregator doesn't have Python version matrix compatibility problems. It runs on a specific version and there's some workaround code for Python 311 in there that just doesn't apply here.
  - there's one weird case where we want this to perform slightly differently around, just not caring about finding real time stamps in the database with test data
- we're only dealing with this one function for now, so the library class is a little heavy and having its own thing provides a nice isolation. If that changed that I might feel more strongly about importing that class.